packages/client/components/GitHubScopingSearchFilterMenu.tsx

```diff
@@ -96,88 +97,67 @@ const GitHubScopingSearchFilterMenu = (props: Props) => {
-      // this might bork if the checkbox is ticked before the full query loads
-      if (!githubSearchQuery) return
-      githubSearchQuery.setValue([], 'nameWithOwnerFilters')
-    })
-  }
   return (
-    <Menu
+    <StyledMenu
       keepParentFocus
       ariaLabel={'Define the GitHub search query'}
       portalStatus={portalStatus}
       isDropdown={isDropdown}
-      resetActiveOnChanges={[selectedAndFilteredProjects]}
     >
-      {isLoading && <MockGitHubFieldList />}
-      {selectedAndFilteredProjects.length > 0 && <FilterLabel>Filter by project:</FilterLabel>}
-      {showSearch && (
-        <SearchItem key='search'>
-          <StyledMenuItemIcon>
-            <SearchIcon>search</SearchIcon>
-          </StyledMenuItemIcon>
-          <TaskFooterIntegrateMenuSearch
-            placeholder={'Search GitHub'}
-            value={value}
-            onChange={onChange}
-          />
-        </SearchItem>
-      )}
-      {(query && selectedAndFilteredProjects.length === 0 && !isLoading && (
-        <NoResults key='no-results'>No GitHub Projects found!</NoResults>
-      )) ||
-        null}
-      {selectedAndFilteredProjects.map((project) => {
-        const {id: globalProjectKey, avatar, name} = project
-        const toggleProjectKeyFilter = () => {
+      <SearchItem key='search'>
+        <StyledMenuItemIcon>
+          <SearchIcon>search</SearchIcon>
+        </StyledMenuItemIcon>
+        <MenuSearch placeholder={'Search your GitHub repos'} value={value} onChange={onChange} />
+      </SearchItem>
+      {isLoading && <MockFieldList />}
+      {edges.length === 0 && !isLoading && <NoResults key='no-results'>No repos found!</NoResults>}
+      {selectedAndFilteredProjects.map((repo) => {
+        const {id: repoId, nameWithOwner} = repo
+        const isSelected = nameWithOwnerFilters.includes(nameWithOwner)
+        const handleClick = () => {
           commitLocalUpdate(atmosphere, (store) => {
             const searchQueryId = SearchQueryId.join('github', meetingId)
             const githubSearchQuery = store.get<GitHubSearchQuery>(searchQueryId)!
-            const nameWithOwnerFiltersProxy = githubSearchQuery
-              .getValue('nameWithOwnerFilters')!
-              .slice()
-            const keyIdx = nameWithOwnerFiltersProxy.indexOf(globalProjectKey)
-            if (keyIdx !== -1) {
-              nameWithOwnerFiltersProxy.splice(keyIdx, 1)
-            } else {
-              nameWithOwnerFiltersProxy.push(globalProjectKey)
-            }
-            githubSearchQuery.setValue(nameWithOwnerFiltersProxy, 'nameWithOwnerFilters')
+            const newFilters = isSelected
+              ? nameWithOwnerFilters.filter((name) => name !== nameWithOwner)
+              : nameWithOwnerFilters.concat(nameWithOwner)
+            const queryString = githubSearchQuery.getValue('queryString')
+            githubSearchQuery.setValue(newFilters, 'nameWithOwnerFilters')
```

Comment:
now that we know the queryString includes all the `repository:FOO` qualifiers, `nameWithOwnerFilters` is just duplicate information & we should remove that field from `githubSearchQuery`. That way, the queryString is the single source of truth & we have fewer variables to keep track of! I'll do this in another PR, just FYI that it'll be going away

Reply:
Makes sense! I considered getting rid of it but thought it might be useful incase we wanted to grab the repos without needing to parse the query string. But I agree that it's cleaner to remove it!